src/main/java/net/oauth/jsontoken/JsonToken.java

```diff
@@ -235,7 +239,17 @@ public JsonObject getPayloadAsJsonObject() {
   }
   
   public String getKeyId() {
-    return signer.getKeyId();
+    if (keyId == null && signer == null) {
+      if (header == null) {
+        throw new NullPointerException("JWT has no signer or header");
```

Comment:
key id is an optional header field (unlike "alg" which is required). Can we just return null instead of throwing an exception if we don't have one available?

Reply:
Yeah, changed